ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/preferences/pages/general/IdeGeneralPreferencesViewImpl.java

```diff
@@ -56,11 +59,26 @@ public void setThemes(List<Theme> themes, String currentThemeId) {
     }
   }
 
+  @Override
+  public boolean isAskBeforeClosingTab() {
+    return askBeforeClosingTab.getValue();
+  }
+
+  @Override
+  public void setAskBeforeClosingTab(boolean askBeforeClosingTab) {
+    this.askBeforeClosingTab.setValue(askBeforeClosingTab);
+  }
+
   @UiHandler("themeBox")
   void handleSelectionChanged(ChangeEvent event) {
     themeBox.getSelectedIndex();
     delegate.themeSelected(themeBox.getValue(themeBox.getSelectedIndex()));
   }
 
-  interface AppearanceViewImplUiBinder extends UiBinder<FlowPanel, AppearanceViewImpl> {}
+  @UiHandler("askBeforeClosingTab")
+  void onAskBeforeClosingTabChanged(final ValueChangeEvent<Boolean> event) {
+    delegate.onAskBeforeClosingTabChanged(event.getValue());
+  }
+
+  interface AppearanceViewImplUiBinder extends UiBinder<FlowPanel, IdeGeneralPreferencesViewImpl> {}
```

Comment:
rename `AppearanceViewImplUiBinder` --> `IdeGeneralPreferencesUiBinder` please

Reply:
thank you, miss it